lib/Navigation/Builder.php

```diff
@@ -114,38 +114,25 @@ protected function resolveOptions(array $options): array
     }
 
     /**
-     * @param array|Document|null $activeDocument
-     * @param Document|null $navigationRootDocument
-     * @param string|null $htmlMenuIdPrefix
-     * @param \Closure|null $pageCallback
-     * @param bool|string $cache
-     * @param int|null $maxDepth
-     * @param int|null $cacheLifetime
-     *
-     * @return mixed|\Pimcore\Navigation\Container
+     * @param array $params
+     * @return Container
      *
      * @throws \Exception
      */
-    public function getNavigation($activeDocument = null, $navigationRootDocument = null, $htmlMenuIdPrefix = null, $pageCallback = null, $cache = true, ?int $maxDepth = null, ?int $cacheLifetime = null)
+    public function getNavigation(array $params)
```

Comment:
@aryaantony92 We should add it also as type hint
```suggestion
    public function getNavigation(array $params): Container
```

Reply:
But, we decided to get rid off duplicate info. So, in this case, I think the phpDoc is complete.

Reply:
Yes if you add the type hint here you can remove it from php doc. But I think we should prefer type hints.

Reply:
https://github.com/pimcore/pimcore/blob/e0f4ec4e5f1bdece047aab175bd14e41b9f52053/lib/Navigation/Builder.php#L116 to https://github.com/pimcore/pimcore/blob/e0f4ec4e5f1bdece047aab175bd14e41b9f52053/lib/Navigation/Builder.php#L121

Reply:
> But, we decided to get rid off duplicate info. So, in this case, I think the phpDoc is complete.

Then... why is there a parameter type? ;)

I'm with @blankse that generally real types should be preferred over PhpDoc.

Reply:
```
    /**
     * @param array{
     *     root?: ?Document,
     *     htmlMenuPrefix?: ?string,
     *     pageCallback?: ?callable,
     *     cache?: string|bool,
     *     cacheLifetime?: ?int,
     *     maxDepth?: ?int,
     *     active?: ?Document,
     *     markActiveTrail?: bool
     * } $params
     *
     * @throws \Exception
     */
    public function getNavigation(array $params): Container
```
I think so it would be perfect :) So the Array Keys are also documented. (It's also requiered for PhpStan level 6)

Reply:
@blankse Accepted the change. Thanks!